[DX-955] Add use-case based SDK references across Pub/Sub pages (#3340)
ably/docs: src/pages/docs/pub-sub/guides/data-streaming.mdx

Ably already powers data streaming across many industries and use cases, such as:

- **Live sports and racing telemetry:** Streaming vehicle or player data to fan applications with hundreds of metrics updating multiple times per second.
- **Financial market data:** Distributing realtime price updates for stocks, cryptocurrencies, and other instruments to trading platforms and analytics dashboards.
- **IoT sensor networks:** Aggregating and distributing data from thousands of sensors across industrial facilities, smart cities, or environmental monitoring systems.
- **Live event platforms:** Managing reactions, chat, and activity feeds during concerts, conferences, or sporting events with thousands of simultaneous participants.
- **Fleet and asset tracking:** Realtime position and status updates for vehicles, equipment, or goods in logistics and supply chain applications.

<Aside data-type='note'>

> For building chat and reaction features, use the [Chat SDK](/docs/chat) which provides purpose-built APIs. The optimization patterns described in this guide apply to the underlying Pub/Sub channels that Chat uses. See [product guidance](/docs/platform/products) for a full comparison.

Reviewer: Missing comma before non-restrictive `which`, plus a small tighten:

```suggestion
For building chat and reaction features, use the [Chat SDK](/docs/chat), which provides purpose-built APIs. The optimization patterns in this guide apply to the Pub/Sub channels Chat is built on. See [product guidance](/docs/platform/products) for a full comparison.
```

Author: Keeping the original. "The underlying Pub/Sub channels that Chat uses" is more explicit about the layering than "the Pub/Sub channels Chat is built on" — readers in this guide need to understand that Chat sits on top of Pub/Sub channels, which is what makes the optimization patterns transferable.

Author: `Pub/Sub channels Chat is built on` doesn't convey `chat` is a product based on `pubsub channels`

Reviewer 2: As above.

Author: https://github.com/ably/docs/pull/3340/changes/fe2d1956260957f50ade0818f3ea54c62f772eb0

Providing consumers only need the latest state, and some latency is acceptable, conflation can dramatically reduces both costs and client load.

<Aside data-type='important'>

>   Conflation is unsuitable for scenarios requiring every message, such as [chat applications](/docs/chat) where losing intermediate messages would impact the user experience.

Reviewer 2: We shouldn't be linking out to chat as a product in a guide about Pub/Sub IMO (same for all links on this page).
